File: fr.legrain.bdg.model.osgi/src/fr/legrain/tiers/model/TiersDossier.java
package fr.legrain.tiers.model;

import fr.legrain.tiers.model.TaTiers;

public class TiersDossier {
	private String codeDossier;
	private TaTiers taTiers;
	
	public TiersDossier() {
		
	}
	
	public TiersDossier(String codeDossier, TaTiers taTiers) {
		this.codeDossier = codeDossier;
		this.taTiers = taTiers;
	}
	
	public String getCodeDossier() {
		return codeDossier;
	}
	public void setCodeDossier(String codeDossier) {
		this.codeDossier = codeDossier;
	}
	public TaTiers getTaTiers() {
		return taTiers;
	}
	public void setTaTiers(TaTiers taTiers) {
		this.taTiers = taTiers;
	}
}

File: fr.legrain.espaceclient.webapp.angular/espaceclient-app/src/app/model/tiers-dossier.ts
import { Injectable } from '@angular/core';
import { Adapter } from './adapter';

export class TiersDossier {
    constructor(
            public codeDossier : string,
            public nomTiers : string,
            public codeTiers : string,
            public nomEntreprise : string,
    //public taTiers: Tiers,
    //    public taTiers: any,
    ){}
    
}

@Injectable({
    providedIn: 'root'
})
export class TiersDossierAdapter implements Adapter<TiersDossier> {

  adapt(item: any): TiersDossier {
      console.log(item);
    return new TiersDossier(
      item.codeDossier,
      item.taTiersDTO.nomTiers,
      item.taTiersDTO.codeTiers,
      item.taTiersDTO.nomEntreprise,
//      item.code,
//      item.name,
//      new Date(item.created),
    );
  }
}